compiler-api/src/main/java/com/squareup/anvil/compiler/api/ComponentMergingBackend.kt

```diff
@@ -0,0 +1,10 @@
+package com.squareup.anvil.compiler.api
+
+/** Possible modes of component merging. */
+public enum class ComponentMergingBackend {
+  /** Component merging runs as an IR plugin during kapt stub generation. */
+  IR,
```

Comment:
How about something like...

```kotlin
public enum class AnvilBackend {
  EMBEDDED_BACKEND,
  KSP_BACKEND
}
public enum class ComponentMergingBackend {
  EMBEDDED_MERGING,
  KSP_MERGING
}
```

Reply:
I don't think that's quite right, as "embedded" in this case means two different things - AnalysisHandlerExtension and IR. I was thinking the specificity was better in this case since `AnvilBackend.EMBEDDED` will soon(er than IR) go away.

Reply:
[abef3a1](https://github.com/square/anvil/pull/886/commits/abef3a153795c6ecf96fab909fa36bf6103a2268)

Reply:
I was approaching this as just a question of whether it's executing as part of KGP (ish) or KSP.  Except KGP isn't really correct either.

I have no strong feeling about it, though, and I'm comfortable with the `IR` name.